azurerm/data_source_express_route_circuit.go

```diff
@@ -0,0 +1,204 @@
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"peer_asn": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"vlan_id": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"shared_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"service_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_provider_properties": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service_provider_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"peering_location": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"bandwidth_in_mbps": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"service_provider_provisioning_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"tier": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"family": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceArmExpressRouteCircuitRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := meta.(*ArmClient).StopContext
+	client := meta.(*ArmClient).expressRouteCircuitClient
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Express Route Circuit %q (Resource Group %q) was not found", name, resourceGroup)
+		}
+		return fmt.Errorf("Error making Read request on the Express Route Circuit %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	if properties := resp.ExpressRouteCircuitPropertiesFormat; properties != nil {
+		peerings := flattenExpressRouteCircuitPeerings(properties.Peerings)
+		if err := d.Set("peerings", peerings); err != nil {
+			return err
+		}
+
+		d.Set("service_key", properties.ServiceKey)
+		d.Set("service_provider_provisioning_state", properties.ServiceProviderProvisioningState)
+
+		if serviceProviderProperties := flattenExpressRouteCircuitServiceProviderProperties(properties.ServiceProviderProperties); serviceProviderProperties != nil {
+			if err := d.Set("service_provider_properties", serviceProviderProperties); err != nil {
+				return fmt.Errorf("Error setting `service_provider_properties`: %+v", err)
+			}
+		}
+
+	}
+
+	if resp.Sku != nil {
+		sku := flattenExpressRouteCircuitSku(resp.Sku)
+		if err := d.Set("sku", sku); err != nil {
+			return fmt.Errorf("Error setting `sku`: %+v", err)
+		}
+	}
```

Comment:
(since this is the same in the other resource, it's fine to leave this as-is for now) in general we'd tend to ensure the `sku` block was always set, for example by making this:

```
sku := flattenExpressRouteCircuitSku(resp.Sku)
if err := d.Set("sku", sku); err != nil {
  return fmt.Errorf("Error setting `sku`: %+v", err)
}
```

(e.g. by removing the surrounding `if`)

and then updating the `flattenExpressRouteCircuitSku` method to be:

```
func flattenExpressRouteCircuitSku(sku *network.ExpressRouteCircuitSku) []interface{} {
	if sku == nil {
	  return []interface{}{}
	}

	return []interface{}{
		map[string]interface{}{
			"tier":   string(sku.Tier),
			"family": string(sku.Family),
		},
	}
}
```

(but as mentioned above, since this is the same in the resource this is fine for now, just for info 😄)